spec/unit/resource/inspec_input_spec.rb

```diff
@@ -0,0 +1,300 @@
+#
+# Copyright:: Copyright (c) Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "spec_helper"
+
+describe Chef::Resource::InspecInput do
+  def load_input(filename)
+    path = "/var/chef/cache/cookbooks/acme_compliance/compliance/inputs/#{filename}"
+    run_context.input_collection << Chef::Compliance::Input.from_yaml(events, input_yaml, path, "acme_compliance")
+  end
+
+  let(:node) { Chef::Node.new }
+  let(:events) { Chef::EventDispatch::Dispatcher.new }
+  let(:run_context) do
+    Chef::RunContext.new(node, {}, events).tap do |rc|
+    end
+  end
+  let(:collection) { double("resource collection") }
+  let(:input_yaml) do
+    <<~EOH
+ssh_custom_path: "/whatever2"
+    EOH
+  end
+  let(:input_json) do
+    <<~EOH
+    { "ssh_custom_path": "/whatever2" }
+    EOH
+  end
+  let(:input_toml) do
+    <<~EOH
+ssh_custom_path = "/whatever2"
+    EOH
+  end
+  let(:input_hash) do
+    { ssh_custom_path: "/whatever2" }
+  end
+  let(:resource) do
+    Chef::Resource::InspecInput.new("ssh-01", run_context)
+  end
+  let(:provider) { resource.provider_for_action(:add) }
+
+  before do
+    allow(run_context).to receive(:resource_collection).and_return(collection)
+  end
+
+  it "sets the default action as :add" do
+    expect(resource.action).to eql([:add])
+  end
+
+  context "with a input in a cookbook" do
+    it "enables the input by the name of the cookbook" do
+      load_input("default.yml")
+      resource.name "acme_compliance"
+      resource.run_action(:add)
+      expect(run_context.input_collection.first).to be_enabled
+      expect(resource).not_to be_updated_by_last_action
+    end
+
+    it "enables the input with a regular expression for the cookbook" do
+      load_input("default.yml")
+      resource.name "acme_comp.*"
+      resource.run_action(:add)
+      expect(run_context.input_collection.first).to be_enabled
+      expect(resource).not_to be_updated_by_last_action
+    end
+
+    it "enables the input with an explicit name" do
```

Comment:
I don't think there's a test here for success when cookbook is given and it should load default., name 'acme_compliance' with a default present.